indico/modules/rb_new/client/js/modules/bookRoom/SearchResultCount.jsx

```diff
@@ -0,0 +1,186 @@
+                </Translate>
+            </Popup>
+        );
+    }
+
+    renderUnavailable(count) {
+        const label = <Label color="red" horizontal size="small">{count}</Label>;
+        const trigger = (
+            <Menu.Item link onClick={this.openUnavailableRoomsModal}>
+                <Icon name="remove" />
+                <Translate>
+                    Unavailable
+                    <Param name="count"
+                           value={label} />
+                </Translate>
+            </Menu.Item>
+        );
+        return (
+            <Popup trigger={trigger}>
+                <Translate>
+                    Rooms unavailable during that time slot (click for details).
+                </Translate>
+            </Popup>
+        );
+    }
+
+    renderNoRooms() {
+        return (
+            <Message icon="times circle outline"
+                     error
+                     content={Translate.string('No rooms match your query')} />
+        );
+    }
+
+    renderNoMatching() {
+        return (
+            <Message icon="warning sign"
+                     warning
+                     content={Translate.string('No rooms are available during that timeslot')} />
+        );
+    }
+
+    openUnavailableRoomsModal = () => {
+        const {pushState} = this.props;
+        pushState('/book/unavailable', true);
+    };
+
+    closeModal = () => {
+        const {pushState} = this.props;
+        pushState(`/book`, true);
+    };
+
+    render() {
+        const {isFetching, matching, total} = this.props;
+        const style = {
+            display: (isFetching || (total > 0)) ? 'inline-flex' : 'none'
+        };
+
+        return (
+            <div styleName="results-count">
+                <Menu icon
+                      styleName="results-count-menu"
+                      className={isFetching ? 'loading' : null}
+                      style={style}>
+                    {isFetching ? (
+                        <div styleName="loading-indicator">
+                            <div className="bar" />
+                            <div className="bar" />
+                            <div className="bar" />
+                        </div>
+                    ) : (
+                        !!total && (
+                            <>
+                                {this.renderRoomTotal(total)}
+                                {this.renderRoomAvailable(matching)}
+                                {(matching < total) && this.renderUnavailable(total - matching)}
+                            </>
+                        )
+                    )}
+                </Menu>
+                {!isFetching && (
+                    <>
+                        {total > 0 && matching === 0 && this.renderNoMatching()}
+                        {total === 0 && this.renderNoRooms()}
+                    </>
+                )}
+                <Route exact path="/book/unavailable" render={() => (
```

Comment:
does the dialog do anything useful when you open that url directly without any data?

Reply:
No. I assume there's always a query string.